### PR TITLE
[PYTHON] Add ov::clone_function bindings

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Add binding for ov::clone_function()
 - Add a wrapper to Model object for model_function()

### Tickets:
 - 67443

### CODE REVIEW

**Review by jiwaszki — APPROVED** [review]

**Review by akuporos — APPROVED** [review]

File: src/bindings/python/src/pyopenvino/graph/model.cpp [hunk 1]
```diff
@@ -662,6 +663,17 @@ void regclass_graph_Model(py::module m) {
             :rtype: bool
         )");
 
+    function.def(
+        "clone_model",
```
Comment (AlexeyLebedev1) [inline src/bindings/python/src/pyopenvino/graph/model.cpp:667]: Why would we want to have this in the Model class? I don't think we should misalign it with C++ here, otherway it should be named as `Model.clone()`.  We can overload  `__copy__` for the Model, and allow user to call it like this:
`from copy import copy`
 `model_copy = copy(model)`

**Review by AlexeyLebedev1 — CHANGES_REQUESTED** [review]

File: src/bindings/python/src/pyopenvino/graph/model.cpp (hunk 1 above)
Comment (jiwaszki) [reply src/bindings/python/src/pyopenvino/graph/model.cpp:667]: Ok, that is a fair point (I missed it in the original comment, I am sorry!)

Comment (jiwaszki) [inline src/bindings/python/src/pyopenvino/graph/model.cpp:667]:
```suggestion
        "clone",
```

**Review by jiwaszki — CHANGES_REQUESTED** [review]

File: src/bindings/python/src/pyopenvino/graph/model.cpp (hunk 1 above)
Comment (jiwaszki) [reply src/bindings/python/src/pyopenvino/graph/model.cpp]: Update: already fixed it, let's wait for CIs:)

File: src/bindings/python/src/pyopenvino/graph/model.cpp [hunk 2]
```diff
@@ -656,6 +657,17 @@ void regclass_graph_Model(py::module m) {
             :rtype: bool
         )");
 
+    model.def(
+        "clone",
```
Comment (ilyachur) [inline src/bindings/python/src/pyopenvino/graph/model.cpp:661]: We don't have such method in C++.

**Review by AlexeyLebedev1 — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**